node/bft/src/worker.rs

```diff
@@ -340,17 +356,26 @@ impl<N: Network> Worker<N> {
     ) -> Result<()> {
         // Construct the transmission.
         let transmission = Transmission::Transaction(transaction.clone());
+        // Compute the checksum.
+        let checksum = transaction.to_checksum::<N>()?;
+        // Construct the transmission ID.
+        let transmission_id = TransmissionID::Transaction(transaction_id, checksum);
         // Remove the transaction from the pending queue.
-        self.pending.remove(&transaction_id, Some(transmission.clone()));
+        self.pending.remove(transmission_id, Some(transmission.clone()));
         // Check if the transaction ID exists.
-        if self.contains_transmission(&transaction_id) {
-            bail!("Transaction '{}' already exists.", fmt_id(transaction_id));
+        if self.contains_transmission(transmission_id) {
+            bail!("Transaction '{}.{}' already exists.", fmt_id(transaction_id), fmt_id(checksum).dimmed());
         }
         // Check that the transaction is well-formed and unique.
         self.ledger.check_transaction_basic(transaction_id, transaction).await?;
         // Adds the transaction to the ready queue.
-        if self.ready.insert(&transaction_id, transmission) {
-            trace!("Worker {} - Added unconfirmed transaction '{}'", self.id, fmt_id(transaction_id));
+        if self.ready.insert(transmission_id, transmission) {
+            trace!(
+                "Worker {}.{} - Added unconfirmed transaction '{}'",
+                self.id,
+                fmt_id(transaction_id),
+                fmt_id(checksum)
```

Comment:
```suggestion
                fmt_id(checksum).dimmed()
```
nit: this is dimmed for `Solution`